File: leetcode/src/main/java/array/CountTheNumberOfVowelStringsInRange.java
package array;

import org.junit.jupiter.api.Test;

import java.util.Set;

import static org.junit.jupiter.api.Assertions.*;

/*
    https://leetcode.com/problems/count-the-number-of-vowel-strings-in-range/

    Cho chuỗi words là hai số nguyên l, r. Một chuỗi gọi là vowel-string nếu ký
    tự đầu và cuối của nó đều là nguyên âm.

    Return the number of vowel strings words[i] where i belongs to the inclusive range [left, right].
 */
public class CountTheNumberOfVowelStringsInRange {

    public int vowelStrings(String[] words, int left, int right) {
        final Set<Character> vowels = Set.of('a', 'e', 'o', 'u', 'i');
        int count = 0;
        for (int i = left; i <= right; i++) {
            String word = words[i];
            char start = word.charAt(0);
            char end = word.charAt(word.length() - 1);
            if (vowels.contains(start) && vowels.contains(end)) {
                count++;
            }
        }
        return count;
    }

    @Test
    public void test() {
        assertEquals(2, vowelStrings(new String[]{"are", "amy", "u"}, 0, 2));
        assertEquals(3, vowelStrings(new String[]{"hey", "aeo", "mu", "ooo", "artro"}, 1, 4));
    }

}

File: py-algorithm/strings/count_the_number_of_vowel_strings_in_range.py
import unittest

from math import gcd
from fractions import Fraction
from typing import List
from queue import PriorityQueue
from collections import Counter
import itertools

"""
    https://leetcode.com/problems/count-the-number-of-vowel-strings-in-range/
"""


class Solution(unittest.TestCase):

    # def vowelStrings_OK(self, words: List[str], left: int, right: int) -> int:
    #     vowels = {'a', 'e', 'o', 'u', 'i'}
    #     count = 0
    #     for i in range(left, right + 1):
    #         word = words[i]
    #         start = word[0]
    #         end = word[-1]
    #         if vowels.__contains__(start) and vowels.__contains__(end):
    #             count += 1
    #
    #     return count

    def vowelStrings(self, words: List[str], left: int, right: int) -> int:
        count = 0
        vowels = {'a', 'e', 'o', 'u', 'i'}
        for i in range(left, right + 1):
            if words[i][0] in vowels and words[i][-1] in vowels:
                count += 1

        return count

    def test(self):
        self.assertEqual(2, self.vowelStrings(["are", "amy", "u"], 0, 2))
        self.assertEqual(3, self.vowelStrings(["hey", "aeo", "mu", "ooo", "artro"], 1, 4))


if __name__ == '__main__':
    unittest.main()
